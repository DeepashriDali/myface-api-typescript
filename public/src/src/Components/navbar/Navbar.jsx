import React from "react";
import {Link} from "react-router-dom"
import './NavBar.scss'

export function Navbar (){

    return <nav>
        <h4 className='nav-link logo'>MyFace</h4>
        <Link className='nav-link' to = "/home">Home</Link>
        <Link className='nav-link' to = "/posts">Posts</Link>
        <Link className='nav-link' to = "/users">Users</Link>
    </nav>

}