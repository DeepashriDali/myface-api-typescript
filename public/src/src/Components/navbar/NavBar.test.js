import { render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Navbar } from './Navbar';
import renderer from 'react-test-renderer';

test("Renders logo", () => {
    render(<BrowserRouter> <Navbar /> </BrowserRouter>);
    const logo = screen.getByText(/myface/i);
    expect(logo).toBeInTheDocument();
});

test("should match snapshot", async () => {
    const tree = renderer.create(<BrowserRouter> <Navbar /> </BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
});