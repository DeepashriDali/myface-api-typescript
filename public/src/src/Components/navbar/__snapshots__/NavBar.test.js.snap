// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Array [
  " ",
  <div
    className="head"
  >
    <h4
      className="nav-link logo"
    >
      MyFace
    </h4>
    <button
      onClick={[Function]}
    >
      Menu
    </button>
    <nav
      className="side-nav"
    >
      <a
        className="nav-link"
        href="/home"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="nav-link"
        href="/posts"
        onClick={[Function]}
      >
        Posts
      </a>
      <a
        className="nav-link"
        href="/users"
        onClick={[Function]}
      >
        Users
      </a>
      <a
        className="nav-link"
        href="/users"
        onClick={[Function]}
      >
        + New User
      </a>
    </nav>
  </div>,
  " ",
]
`;
